cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -262,7 +318,8 @@ public void update(Id.DatasetInstance instance, Map<String, String> properties)
    * @throws DatasetNotFoundException if the dataset instance was not found
    * @throws IOException if there was a problem in checking if the namespace exists over HTTP
    */
-  public void drop(Id.DatasetInstance instance) throws Exception {
+  void drop(Id.DatasetInstance instance) throws Exception {
+    authorizationEnforcementService.enforce(instance.toEntityId(), SecurityRequestContext.toPrincipal(), Action.ADMIN);
```

Comment:
Oh, looks like I misunderstood your comment. I thought this was for create. 

So far (for namespaces, apps, artifacts and programs), if an entity exists, any admin/update operations on that entity require ADMIN on that entity, read requires READ, etc. 

However, just like create, I see your argument that drop also "modifies" the parent. I am not completely certain either way. Let's say Alice creates a dataset and has ADMIN rights on it. Bob, Mary, etc have various privileges on the dataset. Alice decides to drop the dataset, so no one can use it anymore - their privileges are also gone. Once could say that it seems like Alice has too much power, so we enforce write on the namespace as well. However, there could be the other argument then, that Alice now needs extra privileges to drop her own dataset. But for that scenario, we could presume that Alice already has that extra privilege (WRITE an namespace), since she was able to create the dataset earlier.